### PR TITLE
Return generic error messages to clients for server-side errors

### LINKED ISSUE #437: Error responses expose internal error context to clients
## Summary

`user_message()` returns the `Display` representation of `TrustedServerError`, which includes internal error context like "Configuration error: ...", "Proxy error: ...". These are returned directly to clients in the response body.

## Refs

- `crates/fastly/src/error.rs` lines 17-18

## Recommendation

Return generic error messages to clients; keep detailed context in server logs only.

## Context

Production readiness audit — see #396

### PR DESCRIPTION
**Pull request by ChristianPavilonis**
## Summary

- Prevent server-side error details (store names, config paths, proxy addresses) from leaking to clients by replacing the `user_message()` passthrough with categorized responses
- Client errors (4xx) retain brief, safe descriptions; server/integration errors (5xx/502/503) return a generic "An internal error occurred"
- Add unit tests asserting the security boundary: all server-error variants must return only the generic message

## Changes

| File | Change |
| ---- | ------ |
| `crates/trusted-server-core/src/error.rs` | Replace `self.to_string()` in `user_message()` with a `match` that surfaces safe messages for client errors and a generic message for server errors. Add doc comments on `BadRequest` (user-visible message warning), `GdprConsent` (why detail is suppressed), and `user_message` trait method (updated semantics). Add two unit tests covering all variants. |

## Closes

Closes #437

## Test plan

- [x] `cargo test --workspace`
- [x] `cargo clippy --all-targets --all-features -- -D warnings`
- [x] `cargo fmt --all -- --check`
- [x] JS tests: `cd crates/js/lib && npx vitest run`
- [x] JS format: `cd crates/js/lib && npm run format`
- [x] Docs format: `cd docs && npm run format`
- [x] WASM build: `cargo build --bin trusted-server-fastly --release --target wasm32-wasip1`
- [x] Manual testing via `fastly compute serve`

## Checklist

- [x] Changes follow [CLAUDE.md](/CLAUDE.md) conventions
- [x] No `unwrap()` in production code — use `expect("should ...")`
- [x] Uses `log` macros (not `println!`)
- [x] New code has tests
- [x] No secrets or credentials committed

### CODE REVIEW

**Review by aram356 — CHANGES_REQUESTED** [review]
@ChristianPavilonis Please resolve conflicts

File: crates/trusted-server-core/src/error.rs [hunk 1]
```diff
@@ -16,6 +16,9 @@ use http::StatusCode;
 #[derive(Debug, Display)]
 pub enum TrustedServerError {
     /// Client-side input/validation error resulting in a 400 Bad Request.
+    ///
+    /// **Note:** `message` is returned to clients in the HTTP response body.
```
Comment (aram356) [inline crates/trusted-server-core/src/error.rs]: ⛏ **nitpick** — Doc note says "message is returned to clients" but it's not returned verbatim — it's formatted via `format!("Bad request: {message}")` in `user_message()`. Consider: *"The `message` field is included in client-facing HTTP responses via `user_message()`. Keep it free of internal implementation details."*
